dask/array/utils.py

```diff
@@ -77,6 +77,38 @@ def meta_from_array(x, ndim=None, dtype=None):
     return meta
 
 
+def compute_meta(func, dtype, *args, **kwargs):
+    args_meta = list(map(meta_from_array, args))
+    kwargs_meta = {k: meta_from_array(v) for k, v in kwargs.items()}
+
+    # todo: look for alternative to this, causes issues when using map_blocks()
+    # with np.vectorize, such as dask.array.routines._isnonzero_vec().
+    if isinstance(func, np.vectorize):
+        meta = func(*args_meta)
+        if hasattr(meta, 'dtype'):
+            return meta.astype(dtype)
+        else:
+            return None
+
+    try:
+        meta = func(*args_meta, **kwargs_meta)
+    except TypeError as e:
+        if ("unexpected keyword argument" in str(e) or
+                "is an invalid keyword for" in str(e)):
```

Comment:
This seems good for now, but I imagine it won't generalize. It's actually unfortunate that there's no exception that handles exclusively wrong arguments.

Reply:
Most of the time this is handled by Python itself, so the error message is quite standard.  NumPy has its own system, hence why there are two.  

It is unfortunate at how informal/messy this is, but I think it should catch 99% of the cases in the wild.